tools/validators/instance_validator/validate/handler.py

```diff
@@ -108,7 +118,32 @@ def RunValidation(filenames: List[str],
       print('\nError generating universe')
       sys.exit(0)
     print('\nStarting config validation...\n')
-    entities = _ValidateConfig(filenames, universe)
+
+    # Check if the filenames are in fact directories.
+    # If they are not directories, but are instead yaml files,
+    # append them to the list.
+    unpacked_files = []
+    for file in filenames:
+      if os.path.isdir(file):
+        unpacked_files.extend(_GetFilepathsFromDir(file))
+      else:
+        unpacked_files.append(file)
+
+    # Because we are recursively walking the tree, we need to check
+    # that they didnt accidentally pass in a directory that contains
+    # a subdirectory they also passed. This will lead to stupid and
+    # unhelpful key errors if left unchecked. Best to do that here
+    # and save the troubleshooting headache. We could just deduplicate
+    # but this is probably the safer route to go.
+    for elem in unpacked_files:
```

Comment:
An alternative to consider is to change unpacked_files to a set instead of manually checking list elements. The _ValidateConfig and Deserialize type hints would need to be changed to Iterable instead of List, but that's fine since we only use the collection of filenames via iteration. It wouldn't be able to report that a folder was included multiple times, but it also wouldn't encounter issues if that happened.